src/PostgREST/Error.hs

```diff
@@ -216,12 +216,17 @@ instance JSON.ToJSON PgError where
   toJSON (PgError _ usageError) = JSON.toJSON usageError
 
 instance JSON.ToJSON SQL.UsageError where
-  toJSON (SQL.ConnectionError e) = JSON.object [
+  toJSON (SQL.ConnectionUsageError e) = JSON.object [
     "code"    .= ConnectionErrorCode00,
     "message" .= ("Database connection error. Retrying the connection." :: Text),
     "details" .= (T.decodeUtf8With T.lenientDecode $ fromMaybe "" e :: Text),
     "hint"    .= JSON.Null]
-  toJSON (SQL.SessionError e) = JSON.toJSON e -- SQL.Error
+  toJSON (SQL.SessionUsageError e) = JSON.toJSON e -- SQL.Error
+  toJSON SQL.PoolIsReleasedUsageError = JSON.object [
+    "code"    .= InternalErrorCode00,
+    "message" .= ("Use of released pool" :: Text),
+    "details" .= JSON.Null,
+    "hint"    .= JSON.Null]
```

Comment:
Is it possible for the user to get this error? AIUI, on `usePool` we try to recover from `PoolIsReleasedUsageError` immediately.

If it's possible, perhaps this momentary error should include a "Retrying.." message and a `503` status to clarify it's temporary.

Reply:
It's not expected to be visible to the user except during shutdown, so not temporary in that sense.

Reply:
I see. In that case maybe we should a have more user-facing error message, like: 

```suggestion
  toJSON SQL.PoolIsReleasedUsageError = JSON.object [
    "code"    .= InternalErrorCode00,
    "message" .= ("The database connection pool is being released. Shutting down.." :: Text),
    "details" .= JSON.Null,
    "hint"    .= JSON.Null]
```